### DISCUSSION TITLE
Calling set state inside build method

### DISCUSSION THREAD

**Discussion by MkamaMakoko** [rrousselGit/riverpod#1954] [Q&A] [answered]
Hello, in new Notifier provider, is it safe to call set state inside build method?

```
@override
MyState build(){
changeValueA();
return MyState(a: 0, b: 0);
}

void changeValueA()=>state=state.copyWith(a: 1);
```
In this code, what will be the initial state when the build method run for the first time?
Thanks

**Comment by rrousselGit** [comment, marked as answer]
It's fine, yes

**Reply by MkamaMakoko** [reply]
What will be the initial state for above example? The return value or the set state value?

**Reply by rrousselGit** [reply]
The return value.
